### PR TITLE
linuxPackages.rtl8812au: 5.9.3.2.20210427 -> 5.15.6.20210629

### PR DESCRIPTION
**Pull request by moni-dz**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fortuneteller2k** [comment]
Result of `nixpkgs-review pr 182177` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_14_hardened.rtl8812au</li>
    <li>linuxKernel.packages.linux_4_19_hardened.rtl8812au</li>
    <li>linuxKernel.packages.linux_4_9.rtl8812au</li>
    <li>linuxKernel.packages.linux_5_10_hardened.rtl8812au</li>
    <li>linuxKernel.packages.linux_5_15_hardened.rtl8812au</li>
    <li>linuxKernel.packages.linux_5_18_hardened.rtl8812au</li>
    <li>linuxKernel.packages.linux_5_4_hardened.rtl8812au</li>
    <li>linuxKernel.packages.linux_hardened.rtl8812au</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.rtl8812au</li>
  </ul>
</details>
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_14.rtl8812au</li>
    <li>linuxKernel.packages.linux_4_19.rtl8812au</li>
    <li>linuxKernel.packages.linux_5_10.rtl8812au</li>
    <li>linuxKernel.packages.linux_5_15.rtl8812au</li>
    <li>linuxKernel.packages.linux_5_18.rtl8812au</li>
    <li>linuxKernel.packages.linux_5_4.rtl8812au</li>
    <li>linuxKernel.packages.linux_latest_libre.rtl8812au</li>
    <li>linuxKernel.packages.linux_libre.rtl8812au</li>
    <li>linuxKernel.packages.linux_lqx.rtl8812au</li>
    <li>linuxKernel.packages.linux_xanmod.rtl8812au</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rtl8812au</li>
    <li>linuxKernel.packages.linux_zen.rtl8812au</li>
  </ul>
</details>

**Comment by PedroHLC** [comment]
@fortuneteller2k can you bump this one again?
Also, maybe suffix the version with latest commit date `"${kernel.version}-5.13.6.20210629.20220822"`.
Can you check if `kernel.isHardened` was to restrict it from `5.4-hardened`, or if it indeed fails for all hardened kernels?

File: pkgs/os-specific/linux/rtl8812au/default.nix [hunk 1]
```diff
@@ -2,19 +2,17 @@
...
+  version = "${kernel.version}-5.13.6.20210629";
 
   src = fetchFromGitHub {
-    owner = "gordboy";
-    repo = "rtl8812au-5.9.3.2";
-    rev = "6ef5d8fcdb0b94b7490a9a38353877708fca2cd4";
-    sha256 = "sha256-czExf4z0nf7XEJ1YnRSB3CrGV6NTmUKDiZjLmrh6Hwo=";
+    owner = "morrownr";
+    repo = "8812au-20210629";
+    rev = "7de980d325ff7a40a67866bb6cf294f327e36fa2";
```
Comment (superherointj) [inline pkgs/os-specific/linux/rtl8812au/default.nix:10]:
```suggestion
    rev = "7aa0e0c74551b4e4f319f4b54e00ccbfe588b7ce";
```

File: pkgs/os-specific/linux/rtl8812au/default.nix [hunk 2]
```diff
@@ -2,19 +2,17 @@
 
 stdenv.mkDerivation rec {
   pname = "rtl8812au";
-  version = "${kernel.version}-5.9.3.2.20210427";
+  version = "${kernel.version}-5.13.6.20210629";
```
Comment (superherointj) [inline pkgs/os-specific/linux/rtl8812au/default.nix:5]:
```suggestion
  version = "${kernel.version}-unstable-2022-08-22";
```

File: pkgs/os-specific/linux/rtl8812au/default.nix [hunk 3]
```diff
@@ -46,10 +44,10 @@ stdenv.mkDerivation rec {
 
   meta = with lib; {
     description = "Driver for Realtek 802.11ac, rtl8812au, provides the 8812au mod";
-    homepage = "https://github.com/gordboy/rtl8812au-5.9.3.2";
+    homepage = "https://github.com/morrownr/8812au-20210629";
     license = licenses.gpl2Only;
     platforms = platforms.linux;
     maintainers = with maintainers; [ fortuneteller2k ];
-    broken = kernel.kernelOlder "4.10" || kernel.kernelAtLeast "5.15" || kernel.isHardened;
+    broken = kernel.kernelOlder "4.10" || kernel.isHardened;
```
Comment (superherointj) [inline pkgs/os-specific/linux/rtl8812au/default.nix:51]:
```suggestion
    broken = kernel.kernelOlder "4.10" || kernel.isHardened;
```
The broken marking here is imprecise. Needs to run nixpkgs-review (without broken), re-catalog the broken packages, and update this.

**Comment by fortuneteller2k** [comment]
Result of `nixpkgs-review pr 182177` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_testing_bcachefs.rtl8812au</li>
  </ul>
</details>
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_14.rtl8812au</li>
    <li>linuxKernel.packages.linux_4_14_hardened.rtl8812au</li>
    <li>linuxKernel.packages.linux_4_19.rtl8812au</li>
    <li>linuxKernel.packages.linux_4_19_hardened.rtl8812au</li>
    <li>linuxKernel.packages.linux_4_9.rtl8812au</li>
    <li>linuxKernel.packages.linux_5_10.rtl8812au</li>
    <li>linuxKernel.packages.linux_5_10_hardened.rtl8812au</li>
    <li>linuxKernel.packages.linux_5_15.rtl8812au</li>
    <li>linuxKernel.packages.linux_hardened.rtl8812au (linuxKernel.packages.linux_5_15_hardened.rtl8812au)</li>
    <li>linuxKernel.packages.linux_5_18.rtl8812au</li>
    <li>linuxKernel.packages.linux_5_18_hardened.rtl8812au</li>
    <li>linuxKernel.packages.linux_5_19.rtl8812au</li>
    <li>linuxKernel.packages.linux_5_4.rtl8812au</li>
    <li>linuxKernel.packages.linux_latest_libre.rtl8812au</li>
    <li>linuxKernel.packages.linux_libre.rtl8812au</li>
    <li>linuxKernel.packages.linux_lqx.rtl8812au</li>
    <li>linuxKernel.packages.linux_xanmod.rtl8812au</li>
    <li>linuxKernel.packages.linux_xanmod_latest.rtl8812au</li>
    <li>linuxKernel.packages.linux_xanmod_tt.rtl8812au</li>
    <li>linuxKernel.packages.linux_zen.rtl8812au</li>
  </ul>
</details>

**Review by PedroHLC — APPROVED** [review]

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 182177` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.hardkernel_4_14.rtl8812au</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.rtl8812au</li>
    <li>linuxPackages_hardkernel_latest.rtl8812au</li>
    <li>linuxPackages_testing_bcachefs.rtl8812au</li>
  </ul>
</details>
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_14.rtl8812au</li>
    <li>linuxPackages_4_14_hardened.rtl8812au (linuxKernel.packages.linux_4_14_hardened.rtl8812au)</li>
    <li>linuxKernel.packages.linux_4_19.rtl8812au</li>
    <li>linuxPackages_4_19_hardened.rtl8812au (linuxKernel.packages.linux_4_19_hardened.rtl8812au)</li>
    <li>linuxKernel.packages.linux_4_9.rtl8812au</li>
    <li>linuxKernel.packages.linux_5_10.rtl8812au</li>
    <li>linuxPackages_5_10_hardened.rtl8812au (linuxKernel.packages.linux_5_10_hardened.rtl8812au)</li>
    <li>linuxPackages.rtl8812au (linuxKernel.packages.linux_5_15.rtl8812au)</li>
    <li>linuxPackages_hardened.rtl8812au (linuxPackages_5_15_hardened.rtl8812au)</li>
    <li>linuxKernel.packages.linux_5_18.rtl8812au</li>
    <li>linuxPackages_5_18_hardened.rtl8812au (linuxKernel.packages.linux_5_18_hardened.rtl8812au)</li>
    <li>linuxPackages_latest.rtl8812au (linuxKernel.packages.linux_5_19.rtl8812au)</li>
    <li>linuxKernel.packages.linux_5_4.rtl8812au</li>
    <li>linuxPackages_latest-libre.rtl8812au (linuxKernel.packages.linux_latest_libre.rtl8812au)</li>
    <li>linuxPackages-libre.rtl8812au (linuxKernel.packages.linux_libre.rtl8812au)</li>
    <li>linuxPackages_lqx.rtl8812au (linuxKernel.packages.linux_lqx.rtl8812au)</li>
    <li>linuxPackages_xanmod.rtl8812au (linuxKernel.packages.linux_xanmod.rtl8812au)</li>
    <li>linuxPackages_xanmod_latest.rtl8812au (linuxKernel.packages.linux_xanmod_latest.rtl8812au)</li>
    <li>linuxPackages_xanmod_tt.rtl8812au (linuxKernel.packages.linux_xanmod_tt.rtl8812au)</li>
    <li>linuxPackages_zen.rtl8812au (linuxKernel.packages.linux_zen.rtl8812au)</li>
  </ul>
</details>

**Review by superherointj — APPROVED** [review]

**Pull request merged**